### DISCUSSION TITLE
Setting auto-width on a column without defining q-td

### DISCUSSION THREAD

**Discussion by anli-xsigns** [quasarframework/quasar#14045] [General - Components / Directives / etc] [answered]
Hello quasar community!

I have a QTable with defined columns. I want to set the max-width of the first column (ID) to e. g. 200px. I experimented with the style property of the column but that gave no results. I found the auto-width property of QTd API but I want to avoid defining the html like in 
https://stackoverflow.com/questions/63589658/id-value-on-q-td-for-quasar-qtable-when-displaying-complex-content-in-table-cell because I want to reuse the QTable code in a component. I didn't find a proper way to set the column to a smaller width. Do you have any ideas? Thanks a lot for your thoughts in advance!

![Schnappschuss_072522_095932_AM](https://user-images.githubusercontent.com/8227996/180727737-0fdadfe4-9525-4aa5-8716-632099ddd007.jpg)

**Comment by MilosPaunovic** [comment, marked as answer]
You could make use of `classes` attribute while defining columns.

```js
{
  name: 'members',
  label: 'Members',
  required: true,
  align: 'left',
  field: (row) => row.members,
  classes: 'q-table--col-auto-width',
  sortable: false,
},
```

**Reply by anli-xsigns** [reply]
Great idea! Works like charm! Thanks a lot! Have a great day 🥳

**Comment by pratik227** [comment]
@anli-xsigns 

Use Style.

```
{
    name: 'name',
    required: true,
    label: 'Dessert (100g serving)',
    align: 'left',
    field: row => row.name,
    format: val => `${val}`,
    sortable: true,
    style: 'width: 200px',
  }
```

https://codepen.io/Pratik__007/pen/oNqGGxK

**Reply by Scanvaegt** [reply]
In your codepen, try having less columns (e.g. only 3 columns), and you will see the there is a problem. The width is not 200px. Even if you change the style to e.g. 'max-width: 50px' the column is much wider.

**Reply by pratik227** [reply]
<img width="432" alt="Screenshot 2022-10-14 at 4 51 29 PM" src="https://user-images.githubusercontent.com/34883558/195835282-d4003ba9-e0be-413e-ad1b-e8c1db883c2a.png">

It's 200px only

**Reply by Scanvaegt** [reply]
Indeed, you are complete right! 
Thanks for the quick reply. 
The center alignment of 'Calories' tricked me 🙂
![image](https://user-images.githubusercontent.com/26097850/195850818-5c520d23-e67b-4439-9f92-7d284157885e.png)
